app/src/main/java/org/mozilla/rocket/home/topsites/ui/AddNewTopSitesViewModel.kt

```diff
@@ -5,24 +5,49 @@ import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import kotlinx.coroutines.launch
-import org.mozilla.rocket.home.topsites.domain.GetTopSitesUseCase
+import org.mozilla.rocket.adapter.DelegateAdapter
+import org.mozilla.rocket.home.topsites.data.RecommendedSitesResult
+import org.mozilla.rocket.home.topsites.domain.GetRecommendedSitesUseCase
+import org.mozilla.rocket.home.topsites.domain.toFixedSite
 
 class AddNewTopSitesViewModel(
-    private val getTopSitesUseCase: GetTopSitesUseCase
+    private val getRecommendedSites: GetRecommendedSitesUseCase
 ) : ViewModel(), TopSiteClickListener {
 
-    private val _topSitesItems = MutableLiveData<List<Site>>()
-    val topSitesItems: LiveData<List<Site>> = _topSitesItems
+    private val _recommendedSitesResult = MutableLiveData<RecommendedSitesUiResult>()
+    val recommendedSitesItems: LiveData<RecommendedSitesUiResult> = _recommendedSitesResult
 
-    fun requestTopSitesList() {
-        getTopSitesList()
+    fun requestRecommendedSitesList() {
+        getRecommendedSitesList()
     }
 
     override fun onTopSiteClicked(site: Site, position: Int) = Unit
 
     override fun onTopSiteLongClicked(site: Site, position: Int): Boolean = false
 
-    private fun getTopSitesList() = viewModelScope.launch {
-        _topSitesItems.value = getTopSitesUseCase()
+    private fun getRecommendedSitesList() = viewModelScope.launch() {
+        _recommendedSitesResult.value = getRecommendedSites().toUiModel()
```

Comment:
we can consider init the sites when viewmodel initialized

Reply:
fixed